app/actions/remote/post.ts

```diff
@@ -235,26 +237,50 @@ export const postActionWithCookie = async (serverUrl: string, postId: string, ac
     }
 };
 
-const processPostsFetched = (serverUrl: string, actionType: string, data: {order: string[]; posts: Post[]; prev_post_id?: string}, fetchOnly = false) => {
-    const order = data.order;
-    const posts = Object.values(data.posts) as Post[];
-    const previousPostId = data.prev_post_id;
+export async function getMissingChannelsFromPosts(serverUrl: string, posts: Post[], fetchOnly = false) {
+    const operator = DatabaseManager.serverDatabases[serverUrl]?.operator;
+    if (!operator) {
+        return {error: `${serverUrl} database not found`};
+    }
 
-    if (!fetchOnly) {
-        const operator = DatabaseManager.serverDatabases[serverUrl]?.operator;
-        if (operator) {
-            operator.handlePosts({
-                actionType,
-                order,
-                posts,
-                previousPostId,
-            });
+    let client: Client;
+    try {
+        client = NetworkManager.getClient(serverUrl);
+    } catch (error) {
+        return {error};
+    }
+
+    const channelIds = await queryAllMyChannelIds(operator.database);
+    const currentUserId = await queryCurrentUserId(operator.database);
+    const promises1: Array<Promise<Channel>> = [];
+    const promises2: Array<Promise<ChannelMembership>> = [];
+
+    posts.forEach((post) => {
+        const id = post.channel_id;
+
+        if (channelIds.indexOf(id) === -1) {
+            promises1.push(client.getChannel(id));
+            promises2.push(client.getChannelMember(id, currentUserId));
```

Comment:
```suggestion
            promises2.push(client.getMyChannelMember(id));
```

By doing this you avoid having to query the `currentUserId` form the database.